zero_bin/rpc/src/native/txn.rs

```diff
@@ -357,3 +398,228 @@ fn prestate_tracing_options(diff_mode: bool) -> GethDebugTracingOptions {
+    let mut prev_jump = None;
+
+    // Contains the previous op.
+    let mut prev_op = "";
+
+    // Call depth of the previous `entry`.  We initialize to 0 as this compares
+    // smaller to 1.
+    let mut prev_depth = 0;
+    // The next available context. Starts at 1. Never decrements.
+    let mut next_ctx_available = 1;
+    // Immediately use context 1;
+    let mut call_stack = vec![(entrypoint_code_hash, next_ctx_available)];
+    next_ctx_available += 1;
+
+    for (step, entry) in structlog_trace.struct_logs.iter().enumerate() {
+        let op = entry.op.as_str();
+        let curr_depth = entry.depth;
+
+        let exception_occurred = prev_entry_caused_exception(prev_op, prev_depth, curr_depth);
+        if exception_occurred {
+            call_stack.pop().expect("Call stack was empty at POP.");
+        }
+
+        debug_assert!(entry.depth as usize <= next_ctx_available);
+        let (code_hash, ctx) = call_stack.last().unwrap();
+        trace!("TX:   {:?}", tx.hash);
+        trace!("STEP: {:?}", step);
+        trace!("STEPS: {:?}", structlog_trace.struct_logs.len());
+        trace!("PREV OPCODE: {}", prev_op);
+        trace!("OPCODE: {}", entry.op.as_str());
+        trace!("CODE: {:?}", code_hash);
+        trace!("CTX:  {:?}", ctx);
+        trace!("EXCEPTION OCCURED: {:?}", exception_occurred);
+        trace!("PREV_DEPTH:  {:?}", prev_depth);
+        trace!("CURR_DEPTH:  {:?}", curr_depth);
+        trace!("{:#?}\n", entry);
+
+        match op {
+            "CALL" | "CALLCODE" | "DELEGATECALL" | "STATICCALL" => {
+                let callee_address = {
+                    // This is the same stack index (i.e. 2nd) for all four opcodes.  See https://ethervm.io/#F1
+                    let mut evm_stack = entry
+                        .stack
+                        .as_ref()
+                        .expect("No stack found in structLog.")
+                        .iter()
+                        .rev();
+                    let callee_raw = *evm_stack
+                        .nth(1)
+                        .expect("Stack must contain at least two values for a CALL instruction.");
+                    let lower_bytes = U160::from(callee_raw);
+                    Address::from(lower_bytes)
+                };
+
+                if precompiles().contains(&callee_address) {
+                    debug!("PRECOMPILE at address {} called.", &callee_address);
+                } else if callee_addr_to_code_hash.contains_key(&callee_address) {
+                    let code_hash = callee_addr_to_code_hash[&callee_address];
+                    call_stack.push((code_hash, next_ctx_available));
+                } else {
+                    // This case happens if calling an EOA. See also https://www.evm.codes/#00?fork=cancun
+                    debug!(
+                        "Callee address {} has no associated `code_hash`.",
+                        &callee_address
+                    );
+                }
```

Comment:
I think it'd be clearer to just silently ignore this case, as we don't expect to perform anything on non-contract targets

Reply:
I'd like to make it explicit, now that it burnt me once.  Downgraded to `trace!`.